docs/user_guide/ratelimiter.md

```diff
@@ -0,0 +1,116 @@
+# **RateLimiter quickstart**
+Pingora provides a crate `pingora-limits` which provides a simple and easy to use rate limiter for your application.
+
+## Steps
+1. Add the following dependencies to your `Cargo.toml`:
+   ```toml
+   pingora-limits = "0.1.0"
+   lazy_static = "1.4.0"
+   ```
+2. Declare a global rate limiter map to store the rate limiter for each client. In this example, we use `appid`.
+3. Override the `request_filter` method in the `ProxyHttp` trait to implement rate limiting.
+   1. Retrieve the client appid from header.
+   2. Retrieve the current window requests from the rate limiter map. If there is no rate limiter for the client, create a new one and insert it into the map.
+   3. If the current window requests exceed the limit, return 429 and set RateLimiter associated headers.
+   4. If the request is not rate limited, return `Ok(false)` to continue the request.
+
+## Example
+```rust
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+use async_trait::async_trait;
+use lazy_static::lazy_static;
+use pingora::http::ResponseHeader;
+use pingora::prelude::*;
+use pingora_limits::rate::Rate;
+
+fn main() {
+   let mut server = Server::new(Some(Opt::default())).unwrap();
+   server.bootstrap();
+   let mut upstreams = LoadBalancer::try_from_iter(["1.1.1.1:443", "1.0.0.1:443"]).unwrap();
+   // Set health check
+   let hc = TcpHealthCheck::new();
+   upstreams.set_health_check(hc);
+   upstreams.health_check_frequency = Some(Duration::from_secs(1));
+   // Set background service
+   let background = background_service("health check", upstreams);
+   let upstreams = background.task();
+   // Set load balancer
+   let mut lb = http_proxy_service(&server.configuration, LB(upstreams));
+   lb.add_tcp("0.0.0.0:6188");
+
+   // let rate = Rate
+   server.add_service(background);
+   server.add_service(lb);
+   server.run_forever();
+}
+
+pub struct LB(Arc<LoadBalancer<RoundRobin>>);
```

Comment:
This isn’t very relevant for the docs, but I’m pretty certain that you don’t need reference counting here. `LB(LoadBalancer<RoundRobin>)` would be sufficient.